requirements.txt

```diff
@@ -9,3 +9,4 @@ psutil>=5
 scipy>=1.0
 sympy>=1.3
 dill>=0.3
+z3-solver>=4.7
```

Comment:
This should be moved to `requirements-dev.txt` instead since this should be an optional dependency

Reply:
Sure, I've made the change.